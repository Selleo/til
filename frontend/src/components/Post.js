import React from 'react'
import Markdown from './Markdown'
import CopyPostURL from './CopyURL'
import PostCategories from './PostCategories'
import UserPostMenu from '../authenticated/UserPostMenu'
import ReactionBar from './ReactionBar'
import { Link, useLocation } from 'react-router-dom'
import TextBlock from './TextBlock'
import { format, parseISO } from 'date-fns'

const Post = props => {
  const { post, userPost, userImage, review } = props
  const location = useLocation()

  let title
  if (location.pathname === '/search') {
    title = <TextBlock value={post.title} />
  } else {
    title = post.title
  }

  const parsed = parseISO(post.createdAt)
  const date = format(parsed, ' dd MMM  hh:mm')

  return (
    <article className="post">
      <div className="post__header">
        <div className="post__details">
          <img
            src={post.author.image || userImage}
            className="user__image"
            alt="author-img"
          />
          <div className="post__text-details">
            <div className="post__author">
              {post.author.firstName} {post.author.lastName}
            </div>
            <div className="post__date">{date}</div>
          </div>
        </div>
        {!review && <CopyPostURL postId={post.id} />}
      </div>
      <div>
        <Link className="post__title" to={`/posts/${post.id}`}>
          {title}
        </Link>
      </div>
      <div className="post__body">
        <Markdown source={post.body} />
      </div>
      <div className="post__footer">
        <PostCategories categories={post.categories} />
        {!review && <ReactionBar post={post} />}
      </div>
      {userPost && <UserPostMenu post={post} />}
    </article>
  )
}

export default Post
