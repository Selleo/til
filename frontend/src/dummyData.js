export const initialState = {
  categories: [],
  currentUser: null,
  users: [],
  posts: []
};

export const user = {
  email: "l.walczak@selleo.com",
  firstName: null,
  image:
    "https://lh5.googleusercontent.com/-4na6W1JJVa0/AAAAAAAAAAI/AAAAAAAAAAA/SDwbLxAgbqc/photo.jpg",
  lastName: null,
  uuid: "36528600-e215-49bf-8d5b-e1096b63c876"
};

export const allUsers = [
  {
    email: "l.walczak@selleo.com",
    firstName: null,
    image:
      "https://lh5.googleusercontent.com/-4na6W1JJVa0/AAAAAAAAAAI/AAAAAAAAAAA/SDwbLxAgbqc/photo.jpg",
    lastName: null,
    uuid: "36528600-e215-49bf-8d5b-e1096b63c876"
  },
  {
    email: "maciekPlacek@selleo.com",
    firstName: null,
    image:
      "https://lh5.googleusercontent.com/-4na6W1JJVa0/AAAAAAAAAAIdsdfjdkjdfi/photo.jpg",
    lastName: null,
    uuid: "36528600-e215-49bf-8d5b-dadlfj33343443"
  }
];

export const categories = [
  { id: 1, name: "java" },
  { id: 2, name: "javascript" }
];

export const posts = [
  {
    author: {
      email: "l.walczak@selleo.com",
      firstName: null,
      image:
        "https://lh5.googleusercontent.com/-4na6W1JJVa0/AAAAAAAAAAI/AAAAAAAAAAA/SDwbLxAgbqc/photo.jpg",
      lastName: null,
      uuid: "36528600-e215-49bf-8d5b-e1096b63c876"
    },
    body: "adsfdf",
    categoriesIds: [2, 1],
    id: 1,
    isPublic: false,
    likes: [
      {
        post_id: 1,
        user_uuid: "36528600-e215-49bf-8d5b-e1096b63c876"
      }
    ],
    likesCount: 1,
    title: "title "
  },
  {
    author: {
      email: "l.walczak@selleo.com",
      firstName: null,
      image:
        "https://lh5.googleusercontent.com/-4na6W1JJVa0/AAAAAAAAAAI/AAAAAAAAAAA/SDwbLxAgbqc/photo.jpg",
      lastName: null,
      uuid: "36528600-e215-49bf-8d5b-e1096b63c876"
    },
    body: "adfefsddf",
    categoriesIds: [3],
    id: 2,
    isPublic: false,
    likes: [],
    likesCount: 0,
    title: "maciekdfd"
  }
];
