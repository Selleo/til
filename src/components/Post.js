import React from "react";

const Post = ({ html, title }) => {
  const createMarkup = html => {
    return { __html: html };
  };
  console.log(html);

  return (
    <section>
      <article>
        <hr />
        <h1>{title}</h1>
        <div className="post" dangerouslySetInnerHTML={createMarkup(html)} />
        <hr />
      </article>
    </section>
  );
};

export default Post;
