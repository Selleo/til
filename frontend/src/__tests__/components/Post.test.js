import React from "react";
import { Provider } from "react-redux";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";
import configureMockStore from "redux-mock-store";
import { render } from "@testing-library/react";
import Post from "../../components/Post";
import { initialState, posts } from "../../dummyData";

describe("<Post/>", () => {
  const history = createMemoryHistory();
  const mockStore = configureMockStore([]);
  const store = mockStore(initialState);
  const props = {
    post: posts[0],
  };
  it("renders properly", () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Router history={history}>
          <Post {...props} />
        </Router>
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
