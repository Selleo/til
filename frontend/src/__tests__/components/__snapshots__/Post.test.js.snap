// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Post/> renders properly 1`] = `
<DocumentFragment>
  <section
    class="post"
  >
    <article>
      <a
        href="/posts/1"
      >
        <h1>
          title 
        </h1>
      </a>
      <p>
        adsfdf
      </p>
    </article>
    <button>
      Copy post Url
    </button>
    <p>
      like: 0
    </p>
    <p>
      funny: 0
    </p>
    <p>
      love: 0
    </p>
    <p>
      surprised: 0
    </p>
  </section>
</DocumentFragment>
`;
